api/data_explorer/controllers/export_url_controller.py

```diff
@@ -132,11 +141,93 @@ def _create_signed_url(gcs_path):
     return signed_url
 
 
+def _get_range_clause(column, value):
+    arr = value.split('-')
+    if len(arr) > 1:
+        low = arr[0]
+        high = arr[1]
+    else:
+        return column + " = " + value
+    if low.endswith('M'):
+        low = int(low[:-1])
+        high = int(high[:-1])
+        low = low * 1000000
+        high = high * 1000000
+    elif low.endswith('B'):
+        low = int(low[:-1])
+        high = int(high[:-1])
+        low = low * 1000000000
+        high = high * 1000000000
+
+    return column + " >= " + str(low) + " AND " + column + " < " + str(high)
+
+
+def _get_clause(column, type, value):
+    """Returns a single condition of a WHERE clause, eg "gender = female"."""
+    if type == 'text':
+        clause = column + ' = "' + value + '"'
+    elif type == 'boolean':
+        clause = column + ' =' + value
+    else:
+        clause = _get_range_clause(column, value)
+    return clause
+
+
+def _get_filter_query(filters):
+    if not filters or not len(filters):
+        return ""
+    facets = current_app.config['UI_FACETS']
+    table_clauses = dict()
+    for filter in filters:
+        arr = filter.split('=')
+        facet = facets[arr[0]]
+        filter_value = arr[1]
+        arr = facet['name'].rsplit('.', 1)
+        table_name = arr[0]
+        column = arr[1]
+        if table_name in table_clauses:
+            table_clauses[table_name].append(
+                _get_clause(column, facet['type'], filter_value))
+        else:
+            table_clauses[table_name] = [
+                _get_clause(column, facet['type'], filter_value)
+            ]
+
+    table_selects = list()
+    primary_key = current_app.config['PRIMARY_KEY']
+    for table_name, clauses in table_clauses.iteritems():
+        table_select = "(SELECT %s FROM `%s` WHERE %s)"
+        where_clause = ""
+        for clause in clauses:
+            if len(where_clause) > 0:
+                where_clause = where_clause + " AND "
+            where_clause = where_clause + clause
+        table_selects.append(
+            table_select % (primary_key, table_name, where_clause))
+
+    query = "SELECT DISTINCT t1.%s FROM " % primary_key
+    cnt = 1
+    for table_select in table_selects:
+        table = "%s t%d" % (table_select, cnt)
+        join = " INNER JOIN %s ON t%d.%s = t%d.%s"
+        if cnt > 1:
+            query = query + join % (table, cnt - 1, primary_key, cnt,
+                                    primary_key)
+        else:
+            query = query + table
+        cnt = cnt + 1
+    return query
+
+
 def export_url_post():  # noqa: E501
     _check_preconditions()
-    entities = _get_entities_dict()
+    data = json.loads(request.data)
+    current_app.logger.info('Request data %s' % request.data)
```

Comment:
nit let's be more specific "Export URL request data %s"

Reply:
Done